tests/internal/aws_s3_local_buffer.c

```diff
@@ -0,0 +1,178 @@
+/* -*- Mode: C; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+
+#include <fluent-bit/flb_sds.h>
+#include <fluent-bit/flb_s3_local_buffer.h>
+#include <fluent-bit/flb_output_plugin.h>
+#include <fluent-bit/flb_mem.h>
+#include <fluent-bit/flb_info.h>
+#include <fluent-bit/flb_aws_util.h>
+
+#include "flb_tests_internal.h"
+
+#define BUFFER_DIRECTORY FLB_TESTS_DATA_PATH "data/s3_local_buffer/"
+#define PLUGIN_NAME "s3_plugin"
+#define TEST_DATA "I love Fluent Bit"
+#define KEY_1 "key1"
+#define KEY_2 "key2"
+#define KEY_3 "key3"
+
+static void check_chunk(struct flb_local_chunk *chunk, char *tag, char *data)
+{
+    int ret;
+    size_t buffer_size;
+    char *buffered_data = NULL;
+    
+    /* Ensure data retreived is same as that which was stored. */
+    TEST_CHECK(strcmp(chunk->tag, tag) == 0);
+    ret = flb_read_file(chunk->file_path, &buffered_data, &buffer_size);
+    TEST_CHECK(ret == 0);
+    TEST_CHECK(strcmp(buffered_data, data) == 0);
+
+    flb_free(buffered_data);
+}
+
+static void test_flb_buffer_put_create_chunk()
+{
+    int ret;
+    char *key1 = KEY_1 ;
+    char *data = TEST_DATA;
```

Comment:
extra space after the semi-colon.. also- why assign local variables in the function to value of the constants? I would just use the values of the constants directly.